### ISSUE TITLE
plotqualityprofile error : Missing x values ?

### ISSUE THREAD

**Issue by sandipansamaddar** [benjjneb/dada2#898]
Hello,
I am trying to analyze ITS sequences following the tutorial but after using cutadapt seems I am ending up with length 0 in some fastqfile. So when I am trying to use `plotQualityProfile` command for the forward and reverse reads it gives me error that missing x values. How can I solve this problem? I have sequenced ITS2 region using primers ITS86F/ITS4R. I am attaching one pre-cutadapted file (SS11R2_R1_filt.fastq.gz) and one cutadapted (SS11R2_R1_cutadapt.fastq.gz) file for your reference. Also the overview of removed sequences are pasted below.

**Overview of removed sequences
length	count	expect	max.err	error counts
3	104	2297.0	0	104
4	1	574.3	0	1
12	3	0.0	1	2 1
13	3	0.0	1	2 1
14	4	0.0	1	2 2
15	1	0.0	1	1
16	9	0.0	1	7 2
17	17	0.0	1	10 7
18	76	0.0	1	13 25 38
19	776	0.0	1	87 666 23
20	145236	0.0	2	141316 3747 173
21	138	0.0	2	33 98 7
22	2	0.0	2	0 1 1
40	1	0.0	2	1
52	1	0.0	2	1**

Thank you.

Sandipan

[SS11R2_R1_cutadapt.fastq.gz](https://github.com/benjjneb/dada2/files/3889199/SS11R2_R1_cutadapt.fastq.gz)
[SS11R2_R1_filt.fastq.gz](https://github.com/benjjneb/dada2/files/3889200/SS11R2_R1_filt.fastq.gz)

**Comment by benjjneb** [comment]
If you remove the empty files (i.e. those with 0 sequences that passed cutadapt) you should be OK.

**Comment by sandipansamaddar** [comment]
Thanks for the reply. So shall I do that with min length option in cutadapt? Or there are other ways for removing reads with 0 sequences.

Thanks,Sandipan



On Monday, November 25, 2019, 4:21 PM, Benjamin Callahan <notifications@github.com> wrote:


If you remove the empty files (i.e. those with 0 sequences that passed cutadapt) you should be OK.

**Comment by benjjneb** [comment]
There is probably a flag to do it in cutadapt (I am not a cutadapt expert).
You can also do it in the shell by checking file size or the number of lines in a files.
Or in R checking the same things.

**Comment by sandipansamaddar** [comment]
Thanks. I figured it out. There were few reads with 0 sequences and those were making the problem. I removed those with cut-adapt. The code I I used :

`for(i in seq_along(fnFs)) {
  system2(cutadapt, args = c(R1.flags, R2.flags, "-n", 2, "-m", (10:10),
                             "-o", fnFs.cut[i], "-p", fnRs.cut[i], 
                             fnFs.filtN[i], fnRs.filtN[i]))
}`

So with the option `-m` I specified 10:10 as the min length in forward and reverse reads.

Thank you again for your help.

Regards,

Sandipan
